base_report_to_printer/models/printing_action.py

```diff
@@ -5,22 +5,22 @@
 # Copyright (C) 2013-2014 Camptocamp (<http://www.camptocamp.com>)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import api, fields, models
+from odoo import fields, models
+
+AVAILABLE_ACTION_TYPES = [
+    ("server", "Send to Printer"),
+    ("client", "Send to Client"),
+    ("user_default", "Use user's defaults"),
+]
 
 
 class PrintingAction(models.Model):
     _name = "printing.action"
     _description = "Print Job Action"
 
-    @api.model
-    def _available_action_types(self):
-        return [
-            ("server", "Send to Printer"),
-            ("client", "Send to Client"),
-            ("user_default", "Use user's defaults"),
-        ]
+    _available_action_types = AVAILABLE_ACTION_TYPES
```

Comment:
I think I liked the previous code better (here, and in all other similar cases), as it allowed for easier extension through overrides.
I wonder, what motivated this change?

Reply:
Hi @ivantodorovich, thanks for your in-depth review

As Odoo native decided to bypass the validation when the selection is a callable in [here](https://github.com/odoo/odoo/pull/154177/commits/eaef3d28236fed7c72da287f060ae37aa8617f01), it breaks some logic of the module, e.g: 
https://github.com/OCA/report-print-send/blob/58cba795f0555c64798507cf3298ac6b05d2f8ee/base_report_to_printer/tests/test_res_users.py#L15-L19

But my approach supports extension as well, no?

Reply:
Could it work with a property? So we keep a method to ease the override

Reply:
Hi @sebalix, using 'property' is a good approach  :+1: , thank you